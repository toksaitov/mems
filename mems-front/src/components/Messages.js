import React from 'react';
import { AnimateGroup } from 'react-animation';

import Message from '../containers/Message.js';

class Messages extends React.Component {
    componentDidMount() {
        this.props.loadMessages();
    }

    componentWillUnmount() {
        this.props.clearError();
    }

    render = () =>
        <AnimateGroup className="row my-3">{
            this.props.messages.map(message => 
                <Message key={message.id} message={message} />
            )
        }</AnimateGroup>;
}

export default Messages;
