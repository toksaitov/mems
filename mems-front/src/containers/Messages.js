import { connect } from 'react-redux';

import { loadMessages } from '../actions/messages.js';
import { clearError } from '../actions/error.js';

import Messages from '../components/Messages.js';

const mapStateToProps = state => ({
    'messages': state.messages
});

const mapDispatchToProps = dispatch => ({
    'loadMessages': () => dispatch(loadMessages()),
    'clearError': () => dispatch(clearError())
});

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Messages);
