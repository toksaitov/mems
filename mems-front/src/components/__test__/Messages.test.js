import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import thunkMiddleware from 'redux-thunk';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';

const middlewares = [thunkMiddleware];
const mockStore = configureStore(middlewares)

import Messages from '../Messages.js';
import MessagesWrapped from '../../containers/Messages.js';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Messages messages={[]} loadMessages={() => {}} />, div);
});

it('renders without and requests to load the message', () => {
    const mockFunction = jest.fn(() => {});

    const div = document.createElement('div');
    ReactDOM.render(<Messages messages={[]} loadMessages={mockFunction} />, div);

    expect(mockFunction.mock.calls.length).toBe(1);
});

it('matches the previous snapshot', () => {
    const initialState = {
        'messages': [
            {
                "id": 1, 
                "content": "hello",
                "createdAt": "2020-05-24T14:01:41.000Z",
                "updatedAt": "2020-05-24T14:01:41.000Z",
                "user":  {
                    "id": 1,
                    "login": "admin"
                }
            }
        ]
    };
    const store = mockStore(initialState)

    const tree = renderer.create(
        <Provider store={store}>
            <MessagesWrapped />
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});

it('shows a specific message', () => {
    const initialState = {
        'messages': [
            {
                "id": 1, 
                "content": "hello",
                "createdAt": "2020-05-24T14:01:41.000Z",
                "updatedAt": "2020-05-24T14:01:41.000Z",
                "user":  {
                    "id": 1,
                    "login": "admin"
                }
            }
        ]
    };
    const store = mockStore(initialState)

    const wrapper = mount(
        <Provider store={store}>
            <MessagesWrapped />
        </Provider>
    );
    expect(wrapper.contains("hello")).toBe(true);
});

it('shows editing buttons for admin', () => {
    const initialState = {
        'messages': [
            {
                "id": 1, 
                "content": "hello",
                "createdAt": "2020-05-24T14:01:41.000Z",
                "updatedAt": "2020-05-24T14:01:41.000Z",
                "user":  {
                    "id": 1,
                    "login": "admin"
                }
            }
        ],
        'user': {
            "id": 1,
            "login": "admin",
            "authorized": true,
            "administrator": true
        }
    };
    const store = mockStore(initialState)

    const wrapper = mount(
        <Provider store={store}>
            <MessagesWrapped />
        </Provider>
    );
    expect(wrapper.find('.edit-button').length).toBe(1);
    expect(wrapper.find('.delete-button').length).toBe(1);
});