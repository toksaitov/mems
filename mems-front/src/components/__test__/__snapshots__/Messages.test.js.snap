// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the previous snapshot 1`] = `
<div
  className="row my-3"
>
  <div
    className="col-md-4"
  >
    <div
      className="message card p-3 m-2 text-white bg-secondary"
    >
      <h5
        className="card-title"
      >
        @
        admin
      </h5>
      <div>
        <div
          className="message-content card-text"
        >
          hello
        </div>
      </div>
      <time
        className="message-date card-subtitle"
      >
        <small>
          <time
            dateTime={"2020-05-24T14:01:41.000Z"}
          >
            May 24, 2020 10:01 AM
          </time>
        </small>
      </time>
    </div>
  </div>
</div>
`;
